lib/ruby_smb/ntlm/client.rb

```diff
@@ -0,0 +1,74 @@
+module RubySMB::NTLM
+  module Message
```

Comment:
I'm surely missing something, but I'm wondering why we don't have a `Message` class that derives from `Net::NTLM::Message` like it is for the `Client` and `Session` classes instead of overriding `deflag` and `serialize ` in a module that extends the object below (`t3.extend(Message)`) ?
```ruby
class Message < Net::NTLM::Message
```